Src/Base/AMReX_GpuFuse.cpp

```diff
@@ -0,0 +1,241 @@
+#include <AMReX_Gpu.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_BLProfiler.H>
+#include <AMReX_OpenMP.H>
+
+namespace amrex {
+namespace Gpu {
+
+#ifdef AMREX_USE_CUDA
+
+namespace {
+    bool s_in_fuse_region = false;
+    // Fusing kernels with elements greater than this are not recommended based tests on v100
+    Long s_fuse_size_threshold = 257*257;
+    // If the number of kernels is less than this, fusing is not recommended based on tests on v100
+    int s_fuse_numkernels_threshold = 3;
+}
+
+std::unique_ptr<Fuser> Fuser::m_instance = nullptr;
+
+Fuser::Fuser ()
+{
+    AMREX_ASSERT(!OpenMP::in_parallel());
+    m_lambda_buf = (char*)The_Pinned_Arena()->alloc(m_nbytes_lambda_buf);
+    m_helper_buf = (FuseHelper*)The_Pinned_Arena()->alloc(m_nhelpers_buf*sizeof(FuseHelper));
+    m_dtor_buf.reserve(1024);
+}
+
+Fuser::~Fuser ()
+{
+    if (m_nlambdas > 0) {
+        Launch();
+    }
+
+    The_Pinned_Arena()->free(m_lambda_buf);
+    The_Pinned_Arena()->free(m_helper_buf);
+}
+
+void Fuser::Launch ()
+{
+    BL_PROFILE("Fuser::Launch()");
+
```

Comment:
I wouldn't leave this timer here. In an actual use case with potentially tons of these calls, I'm not sure what it captures of interest. It might also be deep enough that it could get called an inordinate number of times.

Reply:
For the Castro test I did, it got called 7878 times, but `FillBoundary` got called `17816` times. `Fsmooth` got called 16664 times.  `getFB` got called 35593 times.  I think it is of interest to know how much time we spend in those fused launches.

Reply:
Yeah, I'd like to leave this timer in here for now -- if it gets to be too much we can always remove it later.